docs/4-software-development-practices/4.5.6-sonarqube.md

```diff
@@ -38,20 +38,48 @@ Here we will setup a new SonarQube server which will listen for requests from ou
-3. Go to Jenkins Configure System and configure your SonarQube server.
-![jenkins SonarQube server image](img4/jenkins-sonarqube_server.webp ':class=img-shadow-center :alt= jenkins SonarQube server image')
-4. Modify the Go Unit Testing Freestyle Project to have the SonarQube scanner configured as the first build step.
-![SonarQube build image](img4/sonarqube_first_build.webp ':class=img-shadow-center :alt= SonarQube build image')
-5. You may need to change the Analysis properties to match the structure of the project. Note: you can also set the properties in a sonar-project.properties file in the root of the project.
-6. Build your project and look at the results in SonarQube.
+1. [Sign up for ngrok](https://dashboard.ngrok.com/login)
+2. [Install ngrok](https://ngrok.com/download)
+```zsh
+brew install ngrok/ngrok/ngrok
+```
+3. Collect your [authentication token](https://dashboard.ngrok.com/get-started/your-authtoken) under `Getting Started` >> `Your Authtoken`
+4. Authenticate ngrok in a terminal
+```zsh
+ngrok config add-authtoken <token>
+```
+5. Expose port `9000` with ngrok
+```zsh
+ngrok http 9000
+```
+6. Collect the https url provided by the ngrok output in the terminal (*you will need this for GitHub configuration*)
+> Do not quit out of terminal
 
-# Deliverable
+### Configure the GitHub repo to run SonarQube scanner
+
+> recommendation: continue using your `getting-started` repo from previous chapters for the following section.
+
+1. Create project in SonarQube and setup Github repo for scan:
+  a. Under the `How do you want to create your project?` section in SonarQube, click `< > Manually`
+  ![SonarQube create project](img4/sonarqube-create-project.png ':class=img-shadow-center :alt= SonarQube create project')
+  b. Supply the name of your repo for `Project display name` and `Project key` (only change main branch name if you are not using `main` as default)
+  ![SonarQube name project](img4/sonarqube-name-project.png ':class=img-shadow-center :alt= SonarQube name project')
+  c. Click on the `With GitHub Actions` tutorial
+  ![SonarQube CI image](img4/sonarqube-ci.png ':class=img-shadow-center :alt= SonarQube CI image')
+  d. Follow the directions for `Create GitHub Secrets`, but for the `SONAR_HOST_URL` secret, use the https url provided by ngrok
+  e. Follow the directions for `Create Workflow YAML File`. For the "What option best describes your build?" options, select `Other` (*this will supply you with required file contents to add, but you can rename the workflow and job as you see fit*).
+  f. Add a status badge to your README.md for the scan job
+2. Commit and push changes to your remote repo in GitHub
+3. View actions ran for GitHub repo (*scan should be successful*)
+4. Navigate back to your SonarQube instance and view the projects tab
+5. Click on your project name and review the scan results
```

Comment:
```suggestion
1. Create project in SonarQube and setup Github repo for scan:
  a. Under the `How do you want to create your project?` section in SonarQube, click `< > Manually`
  ![SonarQube create project](img4/sonarqube-create-project.png ':class=img-shadow-center :alt= SonarQube create project')
  
  b. Supply the name of your repo for `Project display name` and `Project key` (only change main branch name if you are not using `main` as default)
  
  ![SonarQube name project](img4/sonarqube-name-project.png ':class=img-shadow-center :alt= SonarQube name project')
  
  c. Click on the `With GitHub Actions` tutorial
  ![SonarQube CI image](img4/sonarqube-ci.png ':class=img-shadow-center :alt= SonarQube CI image')
  
  d. Follow the directions for `Create GitHub Secrets`, but for the `SONAR_HOST_URL` secret, use the https url provided by ngrok
  
  e. Follow the directions for `Create Workflow YAML File`. For the "What option best describes your build?" options, select `Other` (*this will supply you with required file contents to add, but you can rename the workflow and job as you see fit*).
  
  f. Add a status badge to your README.md for the scan job
  
2. Commit and push changes to your remote repo in GitHub

3. View actions ran for GitHub repo (*scan should be successful*)

4. Navigate back to your SonarQube instance and view the projects tab

5. Click on your project name and review the scan results

```

I like a little space in between the numbered steps. Also it is needed for the lettered steps

```suggestion
  d. Follow the directions for `Create GitHub Secrets`, but for the `SONAR_HOST_URL` secret, use the https url provided by ngrok
  
  e. Follow the directions for `Create Workflow YAML File`. For the "What option best describes your build?" options, select `Other` (*this will supply you with required file contents to add, but you can rename the workflow and job as you see fit*).
  
  f. Add a status badge to your README.md for the scan job
  
```

Pretty sure we need some extra spaces here so this renders correctly
![image](https://github.com/liatrio/devops-bootcamp/assets/19497855/57e6cfb4-e831-4498-a055-66638de0120d)